pydantic_ai_slim/pydantic_ai/models/__init__.py

```diff
@@ -580,6 +581,16 @@ def part_end_event(next_part: ModelResponsePart | None = None) -> PartEndEvent |
 
                     yield event
 
+                # Flush any buffered content and stream finalize events
+                for finalize_event in self._parts_manager.finalize():
+                    if isinstance(finalize_event, PartStartEvent):
+                        if last_start_event:
+                            end_event = part_end_event(finalize_event.part)
+                            if end_event:
+                                yield end_event
+                        last_start_event = finalize_event
+                    yield finalize_event
```

Comment:
Should we set `finalize_event.previous_part_kind` like we do above? Could we reuse that same logic instead of duplicating it?